src/logger/UsageLoggers.go

```diff
@@ -0,0 +1,65 @@
+package logger
+
+import (
+	"sync"
+	//Library used for getting environment variables and other useful env things
+	"os"
+	//used for converting the string returned by lookupEnv
+	"strconv"
+)
+
+//name change since helper uses the name "once"
+var onceUsageLoggers sync.Once
+
+type UsageLoggers struct {
+	bricked  bool
+	disabled bool
+}
+
+var usageLoggers *UsageLoggers
+
+func GetUsageLoggers() (*UsageLoggers, error) {
+	//var envError error
+	var parseError error
+	onceUsageLoggers.Do(func() {
+		//lookup returns a false bool is it fails, along with a nil value.
+		//We can ignore this because parsebool will throw an error anyway if this fails
+		envLookup, _ := os.LookupEnv("USAGE_LOGGERS_DISABLE")
+		_bricked, err := strconv.ParseBool(envLookup)
+		parseError = err
+		usageLoggers = &UsageLoggers{
+			bricked:  _bricked,
+			disabled: _bricked,
+		}
+	})
+	return usageLoggers, parseError
+}
+
+/**
+* Enable all usage loggers, except those explicitly disabled.
+ */
+func (obj UsageLoggers) Enable() {
+	obj.disabled = !obj.bricked
+}
+
+/**
+* Disable all usage loggers.
+ */
+func (obj UsageLoggers) Disable() {
+	obj.disabled = true
+}
+
+/**
+* Returns true if usage loggers are generally enabled.
+ */
+func (obj UsageLoggers) IsEnabled() bool {
+	return !obj.disabled
+}
+
+/**
+* Returns url to use by default.
+ */
+func (obj UsageLoggers) UrlByDefault() (string, bool) {
+	url, lookupSuccess := os.LookupEnv("USAGE_LOGGERS_URL")
```

Comment:
`os.Getenv` should be fine here. If `LookupEnv` returned an error I would say use it but since both `Getenv` and `LookupEnv` return an empty value if the environment variable is not present having a bool attached telling us the same thing isn't very valuable. Once we see the value is empty we know either the variable isn't present or wasn't set.

Reply:
thoughts?

Reply:
Sure, that does simplify things a bit